### PR TITLE
PLA-6221 deprecate DWs and bump version

### PR DESCRIPTION
**Pull request by talltea**
# Citrine Python PR

## Description 
Adds deprecation flags to Workflows. Some of the messages point to PEWs as well as DWs. I could use extra eyes 👀 to make sure I didn't miss places that should have warnings.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/workflow.py [hunk 1]
```diff
@@ -28,6 +31,12 @@ class WorkflowCollection(Collection[Workflow]):
     _resource = Workflow
 
     def __init__(self, project_id: UUID, session: Session):
+        msg = "{this_class} is deprecated. Please use {dw_replacement} instead for " \
+            "Design Workflows and {pew_replacement} for Predictor Evaludation Workflows".format(
+                this_class=self.__class__.__name__,
+                dw_replacement=DesignWorkflowCollection.__name__,
+                pew_replacement=PredictorEvaluationWorkflowCollection.__name__)
+        warn(msg)
```
Comment (maxhutch) [inline src/citrine/resources/workflow.py:39]:
```suggestion
        warn(msg, category=DeprecationWarning)
```

File: src/citrine/resources/workflow_executions.py [hunk 2]
```diff
@@ -47,6 +54,12 @@ def __init__(self,
                  session: Optional[Session] = None,
                  version_number: Optional[int] = None,
                  ):
+        msg = "{this_class} is deprecated. Please use {dw_replacement} instead for " \
+            "Design Workflows and {pew_replacement} for Predictor Evaludation Workflows".format(
+                this_class=self.__class__.__name__,
+                dw_replacement=DesignExecution.__name__,
+                pew_replacement=PredictorEvaluationExecution.__name__)
+        warn(msg)
```
Comment (maxhutch) [inline src/citrine/resources/workflow_executions.py:62]:
```suggestion
        warn(msg, category=DeprecationWarning)
```

File: src/citrine/resources/workflow_executions.py [hunk 3]
```diff
@@ -112,6 +125,13 @@ class WorkflowExecutionCollection(Collection[WorkflowExecution]):
 
     def __init__(self, project_id: UUID, workflow_id: Optional[UUID],
                  session: Optional[Session] = None):
+        msg = "{this_class} is deprecated. Please use {dw_replacement} instead for " \
+            "Design Workflows and {pew_replacement} for Predictor Evaludation Workflows".format(
+                this_class=self.__class__.__name__,
+                dw_replacement=DesignExecutionCollection.__name__,
+                pew_replacement=PredictorEvaluationExecutionCollection.__name__)
+        warn(msg)
```
Comment (maxhutch) [inline src/citrine/resources/workflow_executions.py:133]:
```suggestion
        warn(msg, category=DeprecationWarning)
```

File: src/citrine/informatics/workflows/design_workflow.py [hunk 4]
```diff
@@ -108,6 +109,8 @@ def _post_dump(self, data: dict) -> dict:
     @property
     def executions(self) -> WorkflowExecutionCollection:
```
Comment (maxhutch) [inline src/citrine/informatics/workflows/design_workflow.py:110]: We can use the decorator version here, right?  That will make it show up in some IDEs, I think

**Review by maxhutch — CHANGES_REQUESTED** [review]
Blocking request is changing to DeprecationWarnings.  I also think we'll get better use of the decorator for functions.

Is the documentation update going to be a follow-up PR?

**Comment by talltea** [comment]
> Blocking request is changing to DeprecationWarnings. I also think we'll get better use of the decorator for functions.
> 
> Is the documentation update going to be a follow-up PR?

Added DeprecationWarnings (also to PEW where I got the pattern)

Documentation change incoming, I missed that last night

**Review by maxhutch — APPROVED** [review]

File: docs/source/workflows/design_workflows.rst [hunk 5]
```diff
@@ -74,6 +77,58 @@ Materials at the head of the list are the best materials found from searching th
...
+    # To save all candidates in memory in one list:
+    all_candidates = list(execution.candidates())
+
+    # we can confirm the best candidate is at the head of the list using
+    # this candidate will be the same as best_candidate above
+    candidate_with_max_shear_modulus = max(all_candidates, key=lambda candidate: candidate.material.values['Shear modulus'].mean)
+    print(candidate_with_max_shear_modulus)
+
+
+.. warning::
```
Comment (talltea) [inline docs/source/workflows/design_workflows.rst:128]: Added this warning, didn't want to remove the code example all together yet, but moved it below the preferred flow

File: docs/source/workflows/design_workflows.rst [hunk 6]
```diff
@@ -74,6 +77,58 @@ Materials at the head of the list are the best materials found from searching th
 A workflow can be run using the python SDK.
 Triggering a workflow returns a workflow execution object.
 A workflow execution has a status (in progress, succeeded, or failed) and results (once execution has succeeded).
+
+Candidate results are paginated and returned as `DesignCandidate <#design-candidate>`__ objects.
```
Comment (talltea) [inline docs/source/workflows/design_workflows.rst:81]: A better link :D

**Review by maxhutch — APPROVED** [review]

**Pull request merged**